### PR TITLE
Fixes related to HList type inference

### PR DESCRIPTION
**Pull request by odersky**
Two main fixes and some cleanups:

1. Systematically support dependent implicit methods (the test uses them)
2. Drop unsound rule for &-simplification.

### CODE REVIEW

File: tests/run/HLists-nonvariant.scala [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    println(at(ys4, Succ(Succ(Zero))))
+
+    println(zs1)
+    println(at(zs1, Zero))
+    println(at(zs1, Succ(Zero)))
+    println(at(zs1, Succ(Succ(Zero))))
+  }
+}
+
+object TestNonVariant {
```
Comment (senia-psm) [inline tests/run/HLists-nonvariant.scala]: These objects (`Test` and `TestNonVariant`) are identical. Is it intentional?

**Comment by odersky** [comment]
Something seems to be wrong with the CI on this one.

 - The status says pending
 - The CI job page says all test failed
 - The tests log themselves all say "success"

@felixmulder: any ideas what this could be or how to fix it?

**Comment by odersky** [comment]
Also, I noticed that commits before the last one all get a tick, even if they were not tested themselves. That risks being misleading.

**Comment by odersky** [comment]
I pushed a new commit, but now CI won't even start on this one. It seems the previous zombie run prevents the CI from doing anything further here. Will try with a new PR.

**Comment by felixmulder** [comment]
The bot currently sets the status of each commit that it has checked the
CLA for. I could change it to set a status (failed) only if the user hasn't
signed the CLA...But this won't let us know if the bot is really working...

Yes, kill the PR, probably best...I'm currently in bed with a fever. Will
check more on it tomorrow.

On Thu, Mar 2, 2017, 18:12 odersky <notifications@github.com> wrote:

Also, I noticed that commits before the last one all get a tick, even if
they were not tested themselves. That risks being misleading.

—
You are receiving this because you were mentioned.


Reply to this email directly, view it on GitHub
<https://github.com/lampepfl/dotty/pull/2034#issuecomment-283598306>, or mute
the thread

**Comment by odersky** [comment]
Get better soon, Felix!

**Pull request closed without merging**